infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -117,6 +195,23 @@ protected String evalNamespaceName(String workspaceId, Subject currentUser) {
     }
   }
 
+  protected String evalDefaultNamespaceName(Subject currentUser) {
+    boolean isDefaultPredefined =
+        !isNullOrEmpty(defaultNamespaceName) && hasNoPlaceholders(this.defaultNamespaceName);
+
+    if (isDefaultPredefined) {
+      return this.defaultNamespaceName;
+    } else {
+      String tmpNamespaceName = this.defaultNamespaceName;
+      for (String placeholder : NAMESPACE_NAME_PLACEHOLDERS.keySet()) {
+        tmpNamespaceName =
+            tmpNamespaceName.replaceAll(
```

Comment:
Can `tmpNamespaceName` be null? If it can't we should replace the `isNullOrEmpty` check with just `isEmpty`. If it can be null we could have an NPE here.

Reply:
> If it can be null we could have an NPE here.

In the scope of this method - we could.
in the class - we can't. Agree that it's not so clear and it makes sense to refactor it to make things clearer

Reply:
The situation is a bit better since this method requires non-null `defaultNamespaceName` as argument. Feel free to reopen comment if you have more proposals on how to improve it.

Reply:
Makes sense to me, my confusion in the previous commit was due to null checking with `isNullOrEmpty`, which implies that the value could be null. The changes work for me.